app/models/agents/data_output_agent.rb

```diff
@@ -92,15 +109,39 @@ def feed_link
       interpolated['template']['link'].presence || "https://#{ENV['DOMAIN']}"
     end
 
+    def feed_url(options = {})
+      feed_link + Rails.application.routes.url_helpers.
+                  web_requests_path(agent_id: id || '<id>',
+                                    user_id: user_id,
+                                    secret: options[:secret],
+                                    format: options[:format])
```

Comment:
It should be safe, since the `options` here refers to (pseudo) keyword arguments and it is hardcoded as in `format: :xml`.